// modules/user.js
var mongoClient = require("mongodb").MongoClient;
var ObjectID = require('mongodb').ObjectID;
var setting = require("./setting");

function User(user){
    this.Name = user.Name;
    this.Password = user.Password;
    this.Enable = true;
    this.CreateTime = Date.now();
};
module.exports = User;

User.prototype.save = function(user, callback){
    mongoClient.connect(setting.connString, function(err, db){
        if(err) throw err;

        var userCollection = db.collection("user");
        var info = userCollection.findOne({Name:user.Name});

        if(info == null)
            throw "The username already exists.";

        db.collection("user").insert(user);
        db.close();

        callback(true);
    });
}

User.get = function get(id, callback){
    mongoClient.connect(setting.connString, function(err, db){
        if(err) throw err;

        var userCollection = db.collection("user");
        userCollection.findOne({_id: new ObjectID(id)}, function(err, info){
            db.close();

            callback(info);
        });
    });

}

User.get = function get(username, password, callback){
    mongoClient.connect(setting.connString, function(err, db){
        if(err) throw err;

        var userCollection = db.collection("user");
        userCollection.findOne({Name:username, Password:password}, function(err, info){
            db.close();

            callback(info);
        });
    })
}
